test/powershell/engine/Remoting/RoleCapabilityFiles.Tests.ps1

```diff
@@ -0,0 +1,84 @@
+        }
+        else
+        {
+            [string] $RoleCapDirectory = (New-Item -Path "$TestDrive\RoleCapability" -ItemType Directory -Force).FullName
+
+            [string] $GoodRoleCapFile = "$RoleCapDirectory\TestGoodRoleCap.psrc"
+            New-PSRoleCapabilityFile -Path $GoodRoleCapFile -VisibleCmdlets 'Get-Command','Get-Process','Clear-Host','Out-Default','Select-Object','Get-FormatData','Get-Help'
+
+            [string] $BadRoleCapFile = "$RoleCapDirectory\TestBadRoleCap.psrc"
+            New-PSRoleCapabilityFile -Path $BadRoleCapFile -VisibleCmdlets *
+            [string] $BadRoleCapFile = $BadRoleCapFile.Replace('.psrc', 'psbad')
+
+            [string] $PSSessionConfigFile = "$RoleCapDirectory\TestConfig.pssc"
+        }
+    }
+
+    AfterAll {
+
+        if (!$IsWindows)
+        {
+            $global:PSDefaultParameterValues = $originalDefaultParameterValues
+        }
+    }
+
+    It "Verifies missing role capability file error" {
+
+        New-PSSessionConfigurationFile -Path $PSSessionConfigFile -RoleDefinitions @{
+            Administrators = @{ RoleCapabilityFiles = "$RoleCapDirectory\NoFile.psrc" }
+        }
+
+        try
+        {
+            $iss = [initialsessionstate]::CreateFromSessionConfigurationFile($PSSessionConfigFile, { $true })
+            throw 'Should have thrown CouldNotFindRoleCapabilityFile exception'
+        }
+        catch [System.Management.Automation.MethodInvocationException]
+        {
+            ([System.Management.Automation.PSInvalidOperationException] $_.Exception.InnerException).ErrorRecord.FullyQualifiedErrorId | Should Be 'CouldNotFindRoleCapabilityFile'
+        }
+    }
+
+    It "Verifies incorrect role capability file extenstion error" {
+
+        New-PSSessionConfigurationFile -Path $PSSessionConfigFile -RoleDefinitions @{
+            Administrators = @{ RoleCapabilityFiles = "$BadRoleCapFile" }
+        }
+
+        try
+        {
+            $iss = [initialsessionstate]::CreateFromSessionConfigurationFile($PSSessionConfigFile, { $true })
+            throw 'Should have thrown InvalidRoleCapabilityFileExtension exception'
+        }
+        catch [System.Management.Automation.MethodInvocationException]
+        {
+            ([System.Management.Automation.PSInvalidOperationException] $_.Exception.InnerException).ErrorRecord.FullyQualifiedErrorId | Should Be 'InvalidRoleCapabilityFileExtension'
+        }
+    }
+
+    It "Verifies good role capability file" {
+
+        New-PSSessionConfigurationFile -Path $PSSessionConfigFile -RoleDefinitions @{
+            Administrators = @{ RoleCapabilityFiles = "$GoodRoleCapFile" }
+        }
+
+        $iss = [initialsessionstate]::CreateFromSessionConfigurationFile($PSSessionConfigFile, { $true })
+        [powershell] $ps = [powershell]::Create($iss)
+        $result = $ps.AddCommand('Get-Process').AddParameter('Name', 'PowerShell*').Invoke()
+
+        $result.Count | Should Not Be 0
+        $ps.Dispose()
```

Comment:
In fail test case the `Invoke()` will be raise `CommandNotFoundException` exception.
Also the test check nothing because `Get-Process` is always present by default.
With changing the limit to three cmdlets:
```powershell
New-PSRoleCapabilityFile -Path $GoodRoleCapFile -VisibleCmdlets 'Get-Command','Get-Process','Clear-Host'
```
we could do something like:
```powershell
    It "Verifies good role capability file" {

        try {
            New-PSSessionConfigurationFile -Path $PSSessionConfigFile -RoleDefinitions @{
                Administrators = @{ RoleCapabilityFiles = "$GoodRoleCapFile" }
            }

            $iss = [initialsessionstate]::CreateFromSessionConfigurationFile($PSSessionConfigFile, { $true })
            $ps = [powershell]::Create($iss)
            $result1 = $ps.AddCommand('Get-Command').Invoke()
            $result2 = $ps.AddCommand('Get-Command').AddParameter('Name', 'Clear-Host').Invoke()
            $result3 = $ps.AddCommand('Get-Command').AddParameter('Name', 'Get-Process').Invoke()
        } finally {
            if ($ps) { $ps.Dispose() }
        }
        $result1.Count | Should Be 3
        $result2.Name | Should Be "Clear-Host"
        $result3.Name | Should Be "Get-Process"
    }
```